services/natural_language_understanding/v1-model-manager.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright 2020 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function(RED) {
+  const SERVICE_IDENTIFIER = 'natural-language-understanding',
+    NaturalLanguageUnderstandingV1 = require('ibm-watson/natural-language-understanding/v1'),
+    { IamAuthenticator } = require('ibm-watson/auth'),
+    FEATURE = 'nlu-model-mode',
+    DEFAULT_MODE = 'listModels';
+
+  var pkg = require('../../package.json'),
+    serviceutils = require('../../utilities/service-utils'),
+    payloadutils = require('../../utilities/payload-utils'),
+    sAPIKey = null,
+    apikey = '',
+    service = null,
+    endpoint = '',
+    sEndpoint = 'https://gateway.watsonplatform.net/natural-language-understanding/api';
+
+
+  service = serviceutils.getServiceCreds(SERVICE_IDENTIFIER);
+
+  if (service) {
+    sAPIKey = service.api_key || service.apikey;
+    sEndpoint = service.url;
+  }
+
+  RED.httpAdmin.get('/watson-nlu-model-manager-v4/vcap', function(req, res) {
+    res.json(service ? {
+      bound_service: true
+    } : null);
+  });
+
+  function invokeMethod(node, msg) {
+    return new Promise(function resolver(resolve, reject) {
+      let service = node.service;
+      let method = node.config[FEATURE];
+      let params = node.params;
+
+      service[method](params)
+        .then((data) => {
+          let result = data
+          if (data && data.result) {
+            result = data.result;
+          }
+         resolve(result);
+        })
+        .catch((err) => {
+          reject(err);
+        })
+    });
+  }
+
+  function responseForDeleteMode(node, msg) {
+    let feature = node.config[FEATURE];
+    let field = null;
+
+    switch (feature) {
+      case 'deleteModel':
+        msg.payload = 'Successfully deleted model: ' + node.params.modelId;
+        break;
+      default:
+        return false;
+    }
+    return true;
+  }
+
+  function processTheResponse (body, node, msg) {
+    return new Promise(function resolver(resolve, reject) {
+      if (body == null) {
+        return reject('call to watson nlu v1 service failed');
+      } else if (! responseForDeleteMode(node, msg)) {
+        msg.payload = body;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 16 spaces but found 8.](https://app.codacy.com/manual/BetaWorks-NodeRED-Watson/node-red-node-watson/pullRequest?prid=4804371)